plutus-pab/src/Plutus/PAB/Core/ContractInstance/BlockchainEnv.hs

```diff
@@ -109,27 +109,16 @@ insertNewTx oldMap (txi, txValidity) = do
 -- | Go through the transactions in a block, updating the 'BlockchainEnv'
 --   when any interesting addresses or transactions have changed.
 processMockBlock :: InstancesState -> BlockchainEnv -> Block -> Slot -> STM ()
-processMockBlock instancesState BlockchainEnv{beAddressMap, beTxChanges, beCurrentSlot} transactions slot = do
+processMockBlock instancesState BlockchainEnv{beTxChanges, beCurrentSlot} transactions slot = do
   changes <- STM.readTVar beTxChanges
   forM_ changes $ \tv -> STM.modifyTVar tv increaseDepth
   lastSlot <- STM.readTVar beCurrentSlot
   when (slot > lastSlot) $ do
     STM.writeTVar beCurrentSlot slot
   unless (null transactions) $ do
-    addressMap <- STM.readTVar beAddressMap
-    let addressMap' = foldl' (processTx slot) addressMap transactions
-    STM.writeTVar beAddressMap addressMap'
-
     txStatusMap <- STM.readTVar beTxChanges
     txStatusMap' <- foldM insertNewTx txStatusMap (txMockEvent <$> fmap fromOnChainTx transactions)
     STM.writeTVar beTxChanges txStatusMap'
 
     instEnv <- S.instancesClientEnv instancesState
     updateInstances (indexBlock $ fmap fromOnChainTx transactions) instEnv
-
-processTx :: Slot -> AddressMap -> OnChainTx -> AddressMap
-processTx _ addressMap tx = addressMap' where
-  -- TODO: Will be removed in a future issue
-  addressMap' = AddressMap.updateAllAddresses tx addressMap
-  -- TODO: updateInstances
-  -- We need to switch to using 'ChainIndexTx' everyhwere first, though.
```

Comment:
Not sure if that was the right thing to do, but the `updateInstances` if already called in `processMockBlock`, so I removed `processTx`.